CLUE_Rock_Paper_Scissors/advanced/rps_advertisements.py

```diff
@@ -0,0 +1,237 @@
+        super().__init__()
+        if enc_data is not None:
+            self.enc_data = enc_data
+        if round_no is not None:
+            self.round_no = round_no
+        if ack is not None:
+            self.ack = ack
+        if sequence_number is not None:
+            self.sequence_number = sequence_number
+
+
+class RpsKeyDataAdvertisement(Advertisement):
+    """An RPS (broadcast) message.
+       This sends the key to decrypt the previous encrypted choice of the player.
+       This is not connectable and does not elicit a scan response
+       based on defaults in Advertisement parent class.
+       """
+    flags = None
+
+    _PREFIX_FMT = "<B" "BHBH"
+    _DATA_FMT_KEY_DATA = "8s"
+
+    # prefix appears to be used to determine whether an incoming
+    # packet matches this class
+    # The second struct.calcsize needs to include the _DATA_FMT for some
+    # reason I either don't know or can't remember
+    prefix = struct.pack(
+        _PREFIX_FMT,
+        struct.calcsize(_PREFIX_FMT) - 1,
+        MANUFACTURING_DATA_ADT,
+        ADAFRUIT_COMPANY_ID,
+        struct.calcsize("<H" + _DATA_FMT_KEY_DATA),
+        RPS_KEY_DATA_ID
+    )
+    manufacturer_data = LazyObjectField(
+        ManufacturerData,
+        "manufacturer_data",
+        advertising_data_type=MANUFACTURING_DATA_ADT,
+        company_id=ADAFRUIT_COMPANY_ID,
+        key_encoding="<H"
+    )
+
+    sequence_number = ManufacturerDataField(ADAFRUIT_SEQ_ID, "<" + _SEQ_FMT)
+    """Sequence number of the data. Used in acknowledgements."""
+
+    key_data = ManufacturerDataField(RPS_KEY_DATA_ID, "<" + _DATA_FMT_KEY_DATA)
+    round_no = ManufacturerDataField(RPS_ROUND_ID, "<" + _DATA_FMT_ROUND)
+    ack = ManufacturerDataField(RPS_ACK_ID, "<" + _DATA_FMT_ACK)
+    """Round number starting at 1."""
+
+    def __init__(self, *, key_data=None, round_no=None, ack=None, sequence_number=None):
+        """ack must be set to () to send this optional, data-less field."""
+        super().__init__()
+        if key_data is not None:
+            self.key_data = key_data
+        if round_no is not None:
+            self.round_no = round_no
+        if ack is not None:
+            self.ack = ack
+        if sequence_number is not None:
+            self.sequence_number = sequence_number
+
+
+class RpsRoundEndAdvertisement(Advertisement):
+    """An RPS (broadcast) message.
+       This informs other players the round_no is complete.
+       An important side-effect is acknowledgement of previous message.
+       This is not connectable and does not elicit a scan response
+       based on defaults in Advertisement parent class.
+       """
+    flags = None
+
+    _PREFIX_FMT = "<B" "BHBH"
+
+    prefix = struct.pack(
+        _PREFIX_FMT,
+        struct.calcsize(_PREFIX_FMT) - 1,
+        MANUFACTURING_DATA_ADT,
+        ADAFRUIT_COMPANY_ID,
+        struct.calcsize("<H" + _DATA_FMT_ROUND),
+        RPS_ROUND_ID
+    )
```

Comment:
ditto

Reply:
I have migrated to match_prefixes.

Also did the one in the simple version.